### PR TITLE
[Exp PyROOT] Search for Python once again in SearchIntalledSoftware

### PR DESCRIPTION
**Pull request by maxgalli**
When PR #4279 was initially sent on Aug 25th, the search for Python was
moved to RootBuildMacros in order to avoid a build failure due ot the
search for another package and the declaration of one of the new CMake
variables.
This seem not to be a problem anymore (something changed in the
meantime?), so the search for Python can be put back to the original
place.
This could also fix the tests failing with cxx17 due to a mismatch
between Python executable and libraries.

### CODE REVIEW

File: bindings/pyroot_experimental/pythonizations/CMakeLists.txt [hunk 1]
```diff
@@ -110,7 +110,8 @@ configure_file(${SETUP_PY_IN} ${SETUP_PY})
 
 # create egg with PyROOT info
 execute_process(COMMAND bash -c "${PYTHON_EXECUTABLE} -c 'import setuptools; import distutils'; echo $?" OUTPUT_VARIABLE packages_installed)
```
Comment (etejedor) [inline bindings/pyroot_experimental/pythonizations/CMakeLists.txt]: Need to remove `import distutils` as discussed (not a dependency anymore)

**Comment by maxgalli** [comment]
@phsft-bot build with flags -Dpyroot_experimental=OFF

**Comment by maxgalli** [comment]
@phsft-bot build with flags -Dpyroot_experimental=OFF

**Comment by maxgalli** [comment]
@phsft-bot build with flags -Dpyroot_experimental=ON

**Pull request merged**